### PR TITLE
CATROID-1070 REFACTOR ChangeTempoByAction to Kotlin

### PR DESCRIPTION
**Pull request by sahil9001**
Refactoring of `ChangeTempoByAction` class to Kotlin

Ticket : https://jira.catrob.at/projects/CATROID/issues/CATROID-1070

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by sahil9001** [comment]
Hello @wslany sir! I've made a PR regarding an issue can you please review it?

**Review by frit96 — CHANGES_REQUESTED** [review]
Please update the copyright year at the beginning of the file to 2021.
Also please add a new line at the end of the file so that jenkins can successfully run it's test.

File: catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt [hunk 1]
```diff
@@ -20,42 +20,41 @@
...
+    private var scope: Scope? = null
+    private var tempo: Formula? = null
+    override fun update(delta: Float) {
+        try {
+            val newTempo: Int
+            newTempo = if (tempo == null) {
+                Integer.valueOf(0)
+            } else {
+                tempo!!.interpretInteger(scope)
+            }
```
[the remark this answers is not in the document]
Comment (sahil9001) [reply catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt]: @frit96 I've made the requested changes, please have a look!

**Review by frit96 — APPROVED** [review]
Looks good now thanks

File: catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt [hunk 2]
```diff
@@ -20,42 +20,36 @@
...
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.pocketmusic.mididriver.MidiSoundManager
+
+class ChangeTempoByAction : TemporalAction() {
+    private var scope: Scope? = null
```
Comment (Rawbat) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt:33]: The private keyword is not needed here, kotlin assumes it to be automatically private. Consider changing these two lines to:
lateinit var scope: Scope
lateinit var tempo: Formula
This way the variables do not have to be set nullable.

File: catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt [hunk 3]
```diff
@@ -20,42 +20,36 @@
...
+        } catch (interpretationException: InterpretationException) {
+            Log.d(
+                javaClass.simpleName,
+                "Formula interpretation for this specific Brick failed.",
+                interpretationException
+            )
+        }
+    }
+
+    fun setTempo(tempo: Formula?) {
```
Comment (Rawbat) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeTempoByAction.kt:48]: It is not necessary to explicitly declare setters in kotlin. Please remove them.

**Review by Rawbat — CHANGES_REQUESTED** [review]
Thanks! Have a look at the requested changes and also please squash your 3 commits into a single one.

**Review by frit96 — CHANGES_REQUESTED** [review]
You again haven't added a newline at the end of your file so jenkins gets a warning.
Please always look if the jenkins testsytstem works and has no warnings left.

**Review by Rawbat — CHANGES_REQUESTED** [review]
There seems to be a lint warning. Please try if rebasing fixes this.

**Comment by sahil9001** [comment]
@wslany

**Comment by sahil9001** [comment]
@petbuer

**Comment by petbuer** [comment]
Sorry that we missed this! @frit96 will re-review it soon.

**Review by frit96 — APPROVED** [review]
Looks good

**Comment by sahil9001** [comment]
@Rawbat

**Review by Rawbat — APPROVED** [review]
Sorry for the delay and thanks!

**Pull request merged**